jekyll/_cci2/parallelism-faster-jobs.md

```diff
@@ -81,42 +77,30 @@ jobs:
             circleci tests glob "foo/**/*" "bar/**/*" | xargs -n 1 echo
 ```
 
-### Splitting Test Files
-{:.no_toc}
+#### Splitting by Timing Data
 
-The CLI supports splitting tests across machines
-when running parallel builds.
-To do this,
-pass a list of filenames to the `circleci tests split` command.
+The best way to optimize your test suite across a set of parallel executors is to split your tests using timing data. This will ensure the tests are split in the most even way, leading to a shorter overall test time.
 
-    circleci tests split test_filenames.txt
+![Test Splitting]({{ site.baseurl }}/assets/img/docs/test_splitting.png)
 
-The CLI looks up the number of containers,
-along with the current container index.
-Then, it uses deterministic splitting algorithms
-to split the test files across all available containers.
+On each successful run of a test suite, CircleCI saves timings data from the directory specified by the path in the [`store_test_results`]({{ site.baseurl }}/2.0/configuration-reference/#store_test_results) step. This timings data consists of how long each test took to complete per filename or classname, depending on the language you are using.
 
-By default,
-the number of containers is specified by the `parallelism` key.
-You can manually set this
-by using the `--total` flag.
+Note: If you do not use `store_test_results`, there will be no timing data available for splitting your tests.
 
-    circleci tests split --total=4 test_filenames.txt
+To split by test timings, use the `--split-by` flag with the `timings` split type. The available timings data will then be analyzed and your tests will be split across your parallel-running containers as evenly as possible leading to the fastest possible test run time
 
-Similarly,
-the current container index is automatically picked up from environment variables,
-but can be manually set
-by using the `--index` flag.
+    circleci tests glob "**/*.go" | circleci tests split --split-by=timings
 
-    circleci tests split --index=0 test_filenames.txt
+The CLI expects both filenames and classnames to be present in the timing data produced by the testing suite. By default, splitting defaults to filename, but you can specify classnames by using the `--timings-type` flag.
+
+    cat my_java_test_classnames | circleci tests split --split-by=timings --timings-type=classname
+
+If you need to manually store and retrieve timing data, use the [`store_artifacts`]({{ site.baseurl }}/2.0/configuration-reference/#store_artifacts) step.
```

Comment:
It might be best to leave this sentence out, the timing data is a JSON blob with an undocumented format, so it'd be hard for customers to replicate.

Reply:
Something maybe worth adding is that when new tests are added to a test suite we don't have timing data for them, so we assume they take the same time as the 25th percentile test.

Reply:
Ah right yes good points!